### PR TITLE
🐛  fix: #1316 Infinte Redirects

### PR DESCRIPTION
**Pull request by frimpongopoku**
####  Summary / Highlights

- [X] Before redirecting to custom domain, we check if the user is there already

#### Requirements

* [X] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [X] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [X] I've given my PR a meaningful title.
* [X] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix
- [ ] Feature
- [ ] Performance improvement
- [ ] Feature removal
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch or to a feature branch, _not_ the `main` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the
  issue number)
- [ ] All tests are
  passing: 
- [ ] New/updated tests are included

**Other information: Any otherthing we need to know**

### CODE REVIEW

**Review by abdullai-t — APPROVED** [review]

File: src/components/Utils.js [hunk 1]
```diff
@@ -6,6 +6,22 @@ import { STATUS, ACTIONS } from "react-joyride";
...
+export function domainsAreTheSame(url1, url2) {
+  try {
+      let parsedUrl1 = new URL(url1);
+      let parsedUrl2 = new URL(url2);
+
+      // Remove 'www.' prefix if present
+      let domain1 = parsedUrl1.hostname.replace(/^www\./, '');
+      let domain2 = parsedUrl2.hostname.replace(/^www\./, '');
+
+      return domain1 === domain2;
```
Comment (BradHN1) [inline src/components/Utils.js:19]: question: is there ever a lowercase/uppercase issue to worry about?  If so could convert

**Review by BradHN1 — APPROVED** [review]
LGTM. - thanks for quick worik

File: src/components/Utils.js (hunk 1 above)
Comment (BradHN1) [reply src/components/Utils.js:19]: Looks like my browser converts to lowercase, but just want to be sure in general

Comment (frimpongopoku) [reply src/components/Utils.js:19]: Good observation @BradHN1 , but we do not need to worry about lower case... At least not for the hostname. When we convert the url to the JS URL object, it converts the hostname & host to lowercase... only the pathname is left as is.

**Pull request merged**